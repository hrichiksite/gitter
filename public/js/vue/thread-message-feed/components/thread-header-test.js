const mount = require('../../__test__/vuex-mount');
const { default: ThreadHeader } = require('./thread-header.vue');

describe('thread-message-feed thread-header', () => {
  it('matches snapshot', () => {
    const { wrapper } = mount(ThreadHeader);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('close button calls toggleThreadMessageFeed action', () => {
    const { wrapper, stubbedActions } = mount(ThreadHeader);
    wrapper.find({ ref: 'close-button' }).trigger('click');

    expect(stubbedActions.threadMessageFeed.close).toHaveBeenCalledWith(
      expect.anything(),
      undefined,
      undefined
    );
  });
});
