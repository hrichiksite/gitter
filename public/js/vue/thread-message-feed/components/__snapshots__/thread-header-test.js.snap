// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thread-message-feed thread-header matches snapshot 1`] = `
<header>
  <h2
    class="title"
  >
    Thread
  </h2>
   
  <button
    class="close-button"
  >
    <span
      class="icon-wrapper"
    >
      <svg
        viewBox="0 0 30.5 30.5"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M30,30.5c-0.1,0-0.3,0-0.4-0.1L0.1,0.9C0,0.7,0,0.3,0.1,0.1s0.5-0.2,0.7,0l29.5,29.5c0.2,0.2,0.2,0.5,0,0.7
  C30.3,30.5,30.2,30.5,30,30.5z"
        />
         
        <path
          d="M0.5,30.5c-0.1,0-0.3,0-0.4-0.1c-0.2-0.2-0.2-0.5,0-0.7L29.7,0.1c0.2-0.2,0.5-0.2,0.7,0s0.2,0.5,0,0.7L0.9,30.4
  C0.8,30.5,0.6,30.5,0.5,30.5z"
        />
      </svg>
    </span>
  </button>
</header>
`;
